### PR TITLE
Pre-release stream fixes

### PR DESCRIPTION
**Pull request by s-vitaliy**
Part of https://github.com/SneaksAndData/arcane-stream-blob-storage/issues/5

## Scope

Implemented:
- Fixed incorrect `appVersion` in `Chart.yaml`
- Use the more descriptive plural `blob-storage-streams` instead of `blob-storages`
- Make the `sourceCredentialRef` and `targetCredentialRef` optional
- Use the application full name in the stream class name.
- Implement keyed service injections for the source and the target Amazon SDK client injection.
- Fixed incorrect path resolution in the stream.

## Checklist

- [x] GitHub issue exists for this change.
- [x] Unit tests added and they pass.
- [ ] Line Coverage is at least 80%.
- [x] Review requested on `latest` commit.

### CODE REVIEW

File: src/Extensions/StreamContextExtensions.cs [hunk 1]
```diff
@@ -1,24 +1,18 @@
...
     {
         return Sink.ForEachAsync<(IStoragePath, string)>(context.DeleteParallelism, async deleteRequest =>
         {
-            var (sourceRoot, sourceBlobName) = deleteRequest;
-            var res = await blobStorageService.RemoveBlob(sourceRoot.ObjectKey, sourceBlobName);
-            if (!res)
-            {
-                throw new SinkException($"Failed to remove blob {sourceBlobName} from {sourceRoot}");
-            }
+            await action(deleteRequest);
```
Comment (george-zubrienko) [inline src/Extensions/StreamContextExtensions.cs:15]: if you do this, do you even need to map materialized value? (ie just return `deleteRequest`?)

File: src/GraphBuilder/BlobStorageGraphBuilder.cs [hunk 2]
```diff
@@ -16,15 +18,24 @@ namespace Arcane.Stream.BlobStorage.GraphBuilder;
 
 public class BlobStorageGraphBuilder : IStreamGraphBuilder<BlobStorageStreamContext>
 {
-    private readonly IBlobStorageService sourceBlobStorageService;
+    private readonly IBlobStorageListService sourceBlobStorageService;
```
Comment (george-zubrienko) [inline src/GraphBuilder/BlobStorageGraphBuilder.cs:21]:
```suggestion
    private readonly IBlobStorageListService sourceBlobListStorageService;
```

File: src/GraphBuilder/BlobStorageGraphBuilder.cs [hunk 3]
```diff
@@ -41,32 +53,52 @@ public IRunnableGraph<(UniqueKillSwitch, Task)> BuildGraph(BlobStorageStreamCont
...
-            .GetBlobContentAsync(rootPath.ObjectKey, blobPath, data => data)
-            .Map(d => (rootPath, blobPath, d));
+        this.logger.LogInformation("Reading blob content from {BlobPath}", rootPath.Join(blobPath).ToHdfsPath());
+        return this.sourceBlobStorageReader
+            .GetBlobContentAsync(rootPath.ToHdfsPath(), blobPath, data => data)
+            .Map(data =>
+            {
+                if (data == null)
+                {
+                    throw new ProcessingException(rootPath, blobPath);
```
Comment (george-zubrienko) [inline src/GraphBuilder/BlobStorageGraphBuilder.cs:78]: throw or skip?

**Review by george-zubrienko — APPROVED** [review]

File: src/GraphBuilder/BlobStorageGraphBuilder.cs (hunk 3 above)
Comment (s-vitaliy) [reply src/GraphBuilder/BlobStorageGraphBuilder.cs:78]: Better safe then sorry :)

File: src/Extensions/StreamContextExtensions.cs (hunk 1 above)
Comment (s-vitaliy) [reply src/Extensions/StreamContextExtensions.cs:15]: It's required here because the `ForEachAsync` sink returns `Task<Done` instead of `Task`.

**Pull request merged**